### PR TITLE
Implement all functions

### PR DESCRIPTION
**Pull request by ikarienator**
``` js
var fh = require('.');
console.log(fh.fingerprint128asHex('Hello world'));
```

gives:

```
0e117de994db87c3fd419af489b119af
```

``` js
var fh = require('.');
console.log(fh.fingerprint128asUint32Array('Hello world'));
```

gives

```
Uint32Array {
  '0': 2310085039,
  '1': 4248935156,
  '2': 2497415107,
  '3': 236027369 }
```

### CODE REVIEW

**Comment by ikarienator** [comment]
I can build it with cmake, but I cannot make it build with `npm run build`, so it's not tested.

File: farmhash.cc [hunk 1]
```diff
@@ -1,88 +1,118 @@
...
-  NODE_SET_METHOD(exports, "fingerprint32", fingerprint32);
-  NODE_SET_METHOD(exports, "fingerprint32asHex", fingerprint32asHex);
+}  // namespace farmhash_node
+
+void init(v8::Local<v8::Object> exports) {
+  using namespace farmhash_node;
+
+  NODE_SET_METHOD(exports, "fingerprint32", genFP<32, toNumber<32>>);
+  NODE_SET_METHOD(exports, "fingerprint32asHex", genFP<32, toHex<32>>);
+  NODE_SET_METHOD(exports, "fingerprint32asUint32Array", genFP<32, toUint32Array<32>>);
```
Comment (michaelficarra) [inline farmhash.cc]: Why did you choose `Uint32Array` over `Uint8Array`?

**edit:** Though, I don't see why we couldn't have both.

File: farmhash.cc [hunk 2]
```diff
@@ -1,88 +1,118 @@
 #include <node.h>
+#include <functional>
 
 #define NAMESPACE_FOR_HASH_FUNCTIONS farmhash
+
 #include "./deps/farmhash/src/farmhash.cc"
 
-using namespace v8;
+namespace farmhash_node {
```
Comment (michaelficarra) [inline farmhash.cc]: Other way around: `node_farmhash`.

**Comment by michaelficarra** [comment]
:tada: Thanks!

**Pull request merged**